src/main/java/it/uniroma3/mat/extendedset/intset/ImmutableConciseSet.java

```diff
@@ -694,19 +687,26 @@ public static ImmutableConciseSet doIntersection(Iterator<ImmutableConciseSet> s
     return new ImmutableConciseSet(IntBuffer.wrap(retVal.toArray()));
   }
 
-  private static void sort(final WordIterator[] a, final int to, final Comparator<WordIterator> comp)
+  /**
+   * Variation of insertion sort, elements [maxChangedIndex + 1, size) are sorted, elements [0, maxChangedIndex] should
+   * be inserted into that sorted range.
+   */
+  private static void partialSort(
```

Comment:
Also what do you think about insertion sort implementation with `System.arraycopy(...)`?

```java
private static void partialSort(
    final WordIterator[] a,
    final int maxChangedIndex,
    final int size,
    final Comparator<WordIterator> comp
)
{
  for (int i = maxChangedIndex; i >= 0; i--) {
    WordIterator it = a[i];

    int swapPoint = -1;
    for (int j = i + 1; j < size; j++) {
      if (comp.compare(it, a[j]) <= 0) {
        break;
      }

      swapPoint = j;
    }

    if (swapPoint > -1) {
      System.arraycopy(a, i + 1, a, i, swapPoint - i);
      a[swapPoint] = it;
    }
  }
}
```

For small arrays it might be slightly slower, but for bigger arrays it might give profit. For example on my machine for arrays with size 100k elements Implementation with `arraycopy` works almost 2 times faster.
```
# Run complete. Total time: 00:02:44

Benchmark                                    (size)  Mode  Cnt     Score     Error  Units
InsertionBench.insertionRevert                  100    ss   10     0.004 ±   0.001  ms/op
InsertionBench.insertionRevert                 1000    ss   10     0.266 ±   0.023  ms/op
InsertionBench.insertionRevert                10000    ss   10    31.298 ±   0.839  ms/op
InsertionBench.insertionRevert                50000    ss   10   922.868 ±  33.399  ms/op
InsertionBench.insertionRevert               100000    ss   10  3852.066 ± 168.693  ms/op
InsertionBench.insertionWithArrayCopyRevert     100    ss   10     0.006 ±   0.001  ms/op
InsertionBench.insertionWithArrayCopyRevert    1000    ss   10     0.183 ±   0.009  ms/op
InsertionBench.insertionWithArrayCopyRevert   10000    ss   10    17.683 ±   2.626  ms/op
InsertionBench.insertionWithArrayCopyRevert   50000    ss   10   552.672 ±  28.904  ms/op
InsertionBench.insertionWithArrayCopyRevert  100000    ss   10  2392.526 ±  54.569  ms/op
```